streams/runners/storm/layout/src/main/java/com/hortonworks/streamline/streams/layout/storm/KafkaSpoutFluxComponent.java

```diff
@@ -144,7 +144,15 @@ private String addRecordTranslator () {
                 .JSON_KEY_TOPIC}));
         constructorArgs.add(kafkaSource != null ? kafkaSource.getId() : "");
         constructorArgs.add(conf.get(TopologyLayoutConstants.SCHEMA_REGISTRY_URL));
-        constructorArgs.add(conf.get("readerSchemaVersion") != null ? Integer.parseInt((String) conf.get("readerSchemaVersion")) : null);
+
+        // add readerSchemaVersion to constructor arg only when it's not null and not empty
+        Object readerSchemaVersionObj = conf.get("readerSchemaVersion");
+        if (readerSchemaVersionObj != null) {
+            String readerSchemaVersion = (String) readerSchemaVersionObj;
+            if (!readerSchemaVersion.isEmpty()) {
```

Comment:
nit: This can be simplified like below without having multiple ifs
```
String readerSchemaVersion = (String) conf.get("readerSchemaVersion");
if(readerSchemaVersion != null && !readerSchemaVersion.isEMpty()) {
    constructorArgs.add(Integer.parseInt(readerSchemaVersion))
}
```